msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -11,399 +11,921 @@ ms.date: 10/19/2023
-   Teams Toolkit CLI opens a browser window and requests you to sign in to your Microsoft Account.
+    :::image type="content" source="../assets/images/authentication/teams-sso-tabs/scope-added-msg.png" alt-text="Scope added message":::
 
-1. Sign in to your Microsoft account. Teams Toolkit CLI will execute validation and provisions your app on Azure.
+    The new scope you defined displays on the page.
 
-   :::image type="content" source="../assets/images/Copilot/api-based-CLI-provision-me.png" alt-text="Screenshot shows the sign in request and the provision stages in the command prompt window.":::
+    :::image type="content" source="../assets/images/authentication/teams-sso-tabs/scope-added.png" alt-text="Scope added and displayed":::
 
-1. In the command prompt window, enter the following command to preview your app in Teams:
+#### Configure authorized client application
 
-   ```Preview the app: teamsfx preview --env dev```
+1. Move through the **Expose an API** page to the **Authorized client application** section, and select **+ Add a client application**.
 
- A new browser window with Teams web client opens. You can add your app to Teams.
+    :::image type="content" source="../assets/images/authentication/teams-sso-tabs/auth-client-apps.png" alt-text="Authorized client application":::
 
-# [Visual Studio](#tab/visual-studio)
+    The **Add a client application** page appears.
 
-Before you get started, ensure that you install Visual Studio Enterprise 2022 Preview version 17.9.0 Preview 1.0 and install the **Microsoft Teams development tools** under **ASP.NET and web development** workload.
+1. Enter the appropriate Microsoft 365 client ID for the applications that you want to authorize for your app’s web application.
 
-To create an API-based message extension using Teams Toolkit for Visual Studio, follow these steps:
+    :::image type="content" source="../assets/images/authentication/teams-sso-tabs/add-client-app.png" alt-text="Add a client application":::
 
-1. Open **Visual Studio**.
-1. Go to **File** > **New** > **Project...** or **New Project**.
+    > [!NOTE]
+    >
+    > * The Microsoft 365 client IDs for mobile, desktop, and web applications for Teams, Microsoft 365 app, and Outlook are the actual IDs that you must add.
+    > * For a Teams tab app, you need either Web or SPA, as you can't have a mobile or desktop client application in Teams.
 
-1. Search for **Teams** and select **Microsoft Teams App**.
+    1. Select one of the following client IDs:
 
-   :::image type="content" source="../assets/images/Copilot/api-based-me-vs-teams.png" alt-text="Screenshot shows the Microsoft Teams app option in Visual Studio.":::
+       | Use client ID | For authorizing... |
+       | --- | --- |
+       | 1fec8e78-bce4-4aaf-ab1b-5451cc387264 | Teams mobile or desktop application |
+       | 5e3ce6c0-2b1f-4285-8d4b-75ee78787346 | Teams web application |
+       | 4765445b-32c6-49b0-83e6-1d93765276ca | Microsoft 365 web application |
```

Comment:
Only keep Teams client ids (Teams mobile and Teams web). Remove all the other ids.

Reply:
updated and removed other client IDs from the table.